src/python/pants/backend/jvm/tasks/jvm_compile/jvm_compile.py

```diff
@@ -495,6 +495,8 @@ def __init__(self, *args, **kwargs):
     def _missing_deps_finder(self):
         dep_analyzer = JvmDependencyAnalyzer(
             get_buildroot(),
+            # NB: The analyzer doesn't use the distribution for this usecase, so passing the default
+            # is fine
```

Comment:
Thanks for putting this as a comment.

I was wondering if, while not necessary, it might be useful for us to pass a non-default distribution here? I guess it would require additional wiring to figure out the target platform?

Reply:
The issue is that currently there's one of these for the whole task, and we'd need to change it to be keyed off of the distribution or platform. I tried that and there were more strings than needed. It also isn't currently used by the analyzer for the role it plays in the compile task.
We might want to be able to use different dists for the other usages though. But I didn't explore that yet.